hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java

```diff
@@ -191,11 +207,31 @@ private void registerTarget(final Message message, final String virtualHost) {
         final Target target = controllerManagement.findOrRegisterTargetIfItDoesNotexist(thingId, amqpUri);
         LOG.debug("Target {} reported online state.", thingId);
 
-        lookIfUpdateAvailable(target);
+        sendCurrentUpdateCommandToTarget(target);
     }
 
-    private void lookIfUpdateAvailable(final Target target) {
+    private void sendCurrentUpdateCommandToTarget(final Target target) {
+        if (isMultiAssignmentsEnabled()) {
+            sendCurrentActionsAsMultiActionToTarget(target);
+        } else {
+            sendOldestActionToTarget(target);
+        }
+    }
+
+    private void sendCurrentActionsAsMultiActionToTarget(final Target target) {
```

Comment:
eliminate redundancy... align with dispatcher service...

Reply:
Indeed it seems to be a tight relationship between multi assignment parts and "regular" assignments between `AmqpMessageHandlerService` and `AmqpMessageDispatcherService`. Maybe you split logic in handler and dispatcher  and separate multi assignment into two new classes which common logic could be used from both?